services/src/main/java/org/keycloak/protocol/oid4vc/issuance/signing/SigningService.java

```diff
@@ -34,18 +41,64 @@ public abstract class SigningService<T> implements VerifiableCredentialsSigningS
     // values of the type field are defined by the implementing service. Could f.e. the security suite for ldp_vc or the algorithm to be used for jwt_vc
     protected final String type;
 
-    protected SigningService(KeycloakSession keycloakSession, String keyId, String type) {
+    // As the type is not identical to the format, we use the format as a factory to
+    // instantiate provider.
+    protected final Format format;
+
+    protected SigningService(KeycloakSession keycloakSession, String keyId, Format format, String type) {
         this.keycloakSession = keycloakSession;
         this.keyId = keyId;
+        this.format = format;
         this.type = type;
     }
 
+    @Override
+    public String locator() {
+        return format.name();
+    }
+
+    /**
+     * Returns the key stored under kid, or the active key for the given jws algorithm,
+     *
+     * @param kid
+     * @param algorithm
+     * @return
+     */
     protected KeyWrapper getKey(String kid, String algorithm) {
+        // Allow the service to work with the active key if keyId is null
+        // And we still have to figure out how to proceed with key rotation
+        if(keyId==null){
```

Comment:
It might be better to be `if (keyId == null) {`

Reply:
Resolved with [346b41](https://github.com/keycloak/keycloak/commit/346b4151539737e83126f8cc9705b462441c1081)